### PR TITLE
Audit Report or All subscriptions page for administrators and approvers

### PR DESCRIPTION
**Pull request by santokhsingh**
Test cases for audit report

### CODE REVIEW

File: portal-api/test/06_subscriptionsTest.js [hunk 1]
```diff
@@ -934,27 +960,34 @@ describe('/applications/<appId>/subscriptions', function () {
...
             };
         }
+
         before(function (done) {
-            async.each(appList, (appId, callback) => {
-                utils.createApplication(appId, makeAppInfo(appId), devUserId, function () {
-                    utils.addSubscription(appId, devUserId, privateApi, 'unlimited', null, callback);
-                });
-            }, done);
+            addSomeRegistrations( function () {
```
Comment (DonMartin76) [inline portal-api/test/06_subscriptionsTest.js]: Alt-Shift-F please

File: portal-api/test/06_subscriptionsTest.js [hunk 2]
```diff
@@ -934,27 +960,34 @@ describe('/applications/<appId>/subscriptions', function () {
...
+                async.each(appList, (appId, callback) => {
+                    utils.deleteSubscription(appId, devUserId, privateApi, function () {
+                        utils.deleteApplication(appId, devUserId, callback);
+                    });
+                }, done);
+            });
         });
 
-        it('should, as an admin, be possible to get a list of subscriptions', function (done) {
+        it('should be possible as an admin, to get a list of subscriptions', function (done) {
```
Comment (DonMartin76) [inline portal-api/test/06_subscriptionsTest.js]: This still is grammatically incorrect, isn't it? :-D

File: portal-api/test/06_subscriptionsTest.js [hunk 3]
```diff
@@ -934,27 +960,34 @@ describe('/applications/<appId>/subscriptions', function () {
...
             if (utils.isPostgres()) {
                 request.get({
                     url: baseUrl + 'subscriptions?embed=1&no_cache=1',
-                    headers: utils.makeHeaders(adminUserId, 'read_subscriptions')
+                    headers: utils.makeHeaders(adminUserId, READ_SUBS_SCOPE)
```
Comment (DonMartin76) [inline portal-api/test/06_subscriptionsTest.js:991]: Thank you!

File: portal-api/test/06_subscriptionsTest.js [hunk 4]
```diff
@@ -1004,25 +1037,24 @@ describe('/applications/<appId>/subscriptions', function () {
             }
         });
 
-        it('should, return the list ordered by application id', function (done) {
+        it('should, return the list ordered by application name', function (done) {
```
Comment (DonMartin76) [inline portal-api/test/06_subscriptionsTest.js]: Why did you change this test case? Sorting by name sounds like an additional test case?

File: portal-api/test/06_subscriptionsTest.js [hunk 5]
```diff
@@ -1061,7 +1093,163 @@ describe('/applications/<appId>/subscriptions', function () {
...
+                    headers: utils.makeHeaders(adminUserId, 'read_subscriptions')
+                }, function (err, res, body) {
+                    assert.isNotOk(err);
+                    assert.equal(res.statusCode, 501);
+                    done();
+                });
+            }
+        });
+
+        it('should, return the list filter by application name', function (done) {
```
Comment (DonMartin76) [inline portal-api/test/06_subscriptionsTest.js]: comma?

**Review by DonMartin76 — CHANGES_REQUESTED** [review]
You are not testing for the new feature of automatically/implicitly filtering for the api_group for approval users. There should be a positive test case checking that approvers see the appropriate subscriptions, and at least a second one where we check that approvers *do not see* other subscriptions.

File: portal-api/test/06_subscriptionsTest.js (hunk 4 above)
Comment (santokhsingh) [reply portal-api/test/06_subscriptionsTest.js]: In UI we only expose application name and not application id

**Comment by santokhsingh** [comment]
Done, all review comments implemented
